### PR TITLE
Develop improve renderer

### PR DESCRIPTION
**Pull request by dabingnn**

### CODE REVIEW

File: cocos/2d/renderer/CCRenderer.h [hunk 1]
```diff
@@ -32,10 +32,12 @@
 #include "CCGL.h"
 #include <vector>
 #include <stack>
+#include <list>
```
Comment (dumganhar) [inline cocos/2d/renderer/CCRenderer.h]: Why to include `list`? Where is it used?

File: cocos/2d/renderer/CCRenderer.cpp [hunk 2]
```diff
@@ -412,9 +406,8 @@ void Renderer::drawBatchedQuads()
     int startQuad = 0;
 
     //Upload buffer to VBO
-    if(_numQuads <= 0)
+    if(_numQuads <= 0 || 0 == _batchedQuadCommands.size())
```
Comment (dumganhar) [inline cocos/2d/renderer/CCRenderer.cpp]: Please use `_batchedQualCommands.empty()`.

File: cocos/2d/renderer/CCRenderer.cpp [hunk 3]
```diff
@@ -462,31 +455,27 @@ void Renderer::drawBatchedQuads()
     }
 
     //Start drawing verties in batch
-    for(ssize_t i = _firstCommand; i <= _lastCommand; i++)
+    for(auto i = _batchedQuadCommands.begin(); i != _batchedQuadCommands.end(); ++i)
```
Comment (dumganhar) [inline cocos/2d/renderer/CCRenderer.cpp]: Use c++11 range loop. 
`for (const auto& command : _batchQuadCommands) {}`

**Comment by dabingnn** [comment]
@ricardoquesada can you give a review for this pull request? thank you.

**Comment by dabingnn** [comment]
performance comparison
before:
6d470b3a-2a48-49d1-99f7-738f35b65483
after:
606318b0-6f32-48f2-b76e-4a05475e68f3

**Comment by dumganhar** [comment]
The result looks good. Great finding. :+1:

**Comment by ricardoquesada** [comment]
thanks. looks good to me.

**Pull request merged**